File: left_shift_array.py
def left_shift(arr):
    if not arr:
        return arr
    first_element = arr[0]
    for i in range(len(arr) - 1):
        arr[i] = arr[i + 1]
    arr[-1] = first_element
    return arr

arr = list(map(int, input("Enter array elements: ").split()))
shifted_array = left_shift(arr)
print("Array after left shift:", shifted_array)

File: left_shift_array.cpp
#include <iostream>
#include <vector>
using namespace std;

void leftShift(vector<int>& arr) {
    if (arr.empty()) return;
    int first = arr[0];
    for (int i = 0; i < arr.size() - 1; i++) {
        arr[i] = arr[i + 1];
    }
    
    arr[arr.size() - 1] = first;
}

int main() {
    int n;
    cout << "Enter number of elements: ";
    cin >> n;

    vector<int> arr(n);
    cout << "Enter elements: ";
    for (int i = 0; i < n; i++)
        cin >> arr[i];

    leftShift(arr);

    cout << "Left shifted array: ";
    for (int x : arr)
        cout << x << " ";
    cout << endl;

    return 0;
}

